config/initializers/rack_attack.rb

```diff
@@ -2,26 +2,32 @@
 
 # Throttle general requests by IP
 class Rack::Attack
-  throttle("General requests by ip", limit: 300, period: 5.minutes, &:ip)
-
-  protected_routes = [
-    "/registration/contact_details",
-    "/registration/contact_details/change",
-    "/registration/confirm_email",
-    "/registration/confirm_email/change",
+  PROTECTED_ROUTES = [
     "/registration/qualified_teacher_check",
     "/registration/qualified_teacher_check/change",
     "/session/sign_in",
     "/session/sign_in_code",
-  ]
-  throttle("Rate limit external APIs", limit: 5, period: 1.minute) do |request|
-    request.ip if protected_routes.include?(request.path)
+  ].freeze
+
+  # Throttle protected routes by IP (5rpm)
+  throttle("protected routes (hitting external services)", limit: 5, period: 1.minute) do |request|
+    request.ip if PROTECTED_ROUTES.include?(request.path)
   end
 
   # Throttle /csp_reports requests by IP (5rpm)
   throttle("csp_reports req/ip", limit: 5, period: 1.minute) do |req|
     req.ip if req.path == "/csp_reports"
   end
+
+  # Throttle /api requests by auth token (1000 requests per 5 minutes)
+  throttle("API requests by auth token", limit: 1000, period: 5.minutes) do |request|
+    if request.path.starts_with?("/api/")
+      request.get_header("HTTP_AUTHORIZATION")
+    end
```

Comment:
Good spot; I've added a rule to rate limit this by IP address; previously it would have been caught by the general 300 per 5 minute rule but I've kept it consistent with our 1000 per 5 minute rule for other API requests.